### ISSUE TITLE
Sherpa 2.2.3 compilation problem

### ISSUE THREAD

**Issue by nhartland** [davidchall/homebrew-hep#108]
I'm seeing this when attempting to `brew install sherpa`

```
Last 15 lines from /Users/hartland/Library/Logs/Homebrew/sherpa/02.make:
    ( sizeof(x) == sizeof(float)  ? __inline_isnanf((float)(x))          \
    ^
In file included from STL_Tools.C:4:
In file included from ../../ATOOLS/Phys/Flavour.H:8:
../../ATOOLS/Math/MathTools.H:122:32: error: expected unqualified-id
    return std::isnan(x)||std::isnan(-x);
                               ^
/usr/include/math.h:179:5: note: expanded from macro 'isnan'
    ( sizeof(x) == sizeof(float)  ? __inline_isnanf((float)(x))          \
    ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make[2]: *** [libToolsOrg_la-STL_Tools.lo] Error 1
make[1]: *** [install-recursive] Error 1
make: *** [install-recursive] Error 1
```

Has anyone else noticed this? Oddly enough, if I try to install manually to try and debug I actually get a different bunch of errors around ambiguous abs calls (which is probably a separate issue ticket).

**Comment by ebothmann** [comment]
I had the same thing, but only after upgrading to High Sierra. I have found some similar issues with other brew packages after searching the web.

Are the ambiguous abs calls errors or warnings? I have the feeling clang has upgraded some warnings to errors. However it's definitely a separate thing and easily fixed.

Now if you'd tell me you have not updated to High Sierra I would be utterly confused. Please don't ;)

**Comment by nhartland** [comment]
> Now if you'd tell me you have not updated to High Sierra I would be utterly confused. Please don't ;)

I'm very happy not to disappoint, I upgraded to High Sierra a couple of days ago!

> Are the ambiguous abs calls errors or warnings? I have the feeling clang has upgraded some warnings to errors. However it's definitely a separate thing and easily fixed.

They're errors, and they do appear easy to fix (particularly as lots of them are taking the abs of an unsigned int!). Some of them are a bit more complicated though, and I wouldn't be confident myself in just casting the arguments to an int or whatever to remove the ambiguity by patching.

**Comment by ebothmann** [comment]
Here is another homebrew formula with the same issue stating it's a homebrew problem:
https://github.com/OSGeo/homebrew-osgeo4mac/issues/276

Also see the main thread:
https://github.com/Homebrew/homebrew-core/issues/14418

There seem to be missing files in Xcode 9 CLT. Hopefully this gets fixed soon.

**Comment by davidchall** [comment]
You might try the adding the following line after `def install`:

    ENV.delete("SDKROOT") if DevelopmentTools.clang_build_version >= 900

**Comment by ebothmann** [comment]
Thanks for pointing this out, David. And thanks for merging #109 with a35d389. I think we can close this issue, then.

**Comment by nhartland** [comment]
I just gave the new formula a try, it's working great for me. Thanks guys, much appreciated.
